src/System.Runtime.Extensions/tests/System/Runtime/ProfileOptimization.netcoreapp.cs

```diff
@@ -17,7 +17,15 @@ public void ProfileOptimization_CheckFileExists()
 
             RemoteInvoke(profileFilePath =>
             {
-                ProfileOptimization.SetProfileRoot(Path.GetDirectoryName(profileFilePath));
+                string directoryName = Path.GetDirectoryName(profileFilePath);
+
+                // after test fail tracked by https://github.com/dotnet/corefx/issues/31792
+                // we suspect that the reason is something related to write permission to the location
+                // to prove that we added a simple write to file in same location of profile file directory path
+                // ProfileOptimization/Multi-Core JIT could fail silently
+                File.WriteAllText(Path.Combine(directoryName, Path.GetRandomFileName()), "42");
```

Comment:
@brianrob added write test with comment before start profiling.

Reply:
@MarcoRossignoli thanks for adding this.  Can you also write the full path to the test file to the console?  That way, if it does fail, we will know exactly where it tried to write to.